### PR TITLE
Double IQN

### PR DESCRIPTION
**Pull request by prabhatnagarajan**
This PR adds Double IQN. This PR includes a new example script for IQN based agents (since it does not fall under "reproducibility". It includes a new Agent `DoubleIQN`, as well as a test.

### CODE REVIEW

**Comment by prabhatnagarajan** [comment]
| Game        | Double 3-Step IQN | ChainerRL IQN |
| ------------- |:-------------:|:-------------:|
| Asterix | 402856.2|  **454846.7** |
| BeamRider | 25382.4| **38111.4**|
| Berzerk | 2013.2|**138167.9** |
| Bowling | **90.4**|84.3 |
| Breakout | 393.2| 658.6|
| Enduro | 2362.5| **2363.3**|
| Freeway | **34.0**| **34.0** |
| Hero | **36409.1**| 27639.9|
| MsPacman | **6924.1**| 5786.5 |
| Pong | **21.0**| **21.0**|
| PrivateEye | 200.0|**289.9** |
| Qbert | **27683.3**| 24950.3|
| Seaquest | 9944.8| **27045.5**|
| SpaceInvaders | **53172.5**|27810.9 |
| Venture | **1520.0**|289.0 |
| VideoPinball | **774232.6**|664013.5 |

**Comment by prabhatnagarajan** [comment]
| Game        | CRL IQN | CRL DoubleIQN|
| ------------- |:-------------:|:-------------:|
| Asterix | 454846.7| **483300**|
| Asteroids | **3885.9**| 3860.71|
| Atlantis | 946912.5| **978200**|
| Berzerk | **138167.9**| 2339.0|
| Breakout | **658.6**| 608.3|
| Enduro | **2363.3**| 2359.25|
| Freeway | **34.0**| **34.0**|
| Hero | 27639.9| **28818.65**|
| SpaceInvaders | 27810.9| **54967.5**|

Double IQN wins on 4 domains
IQN wins on 4 domains
Tie on 1 domain

**Comment by prabhatnagarajan** [comment]
Full fledged IQN Results
| Game        | DoubleIQN | ChainerRL IQN |
| ------------- |:-------------:|:-------------:|
| Asterix | **507353.8**|  454846.7 |
|Asteroids| **3915.47** | 3885.9|
| BeamRider | 33666.2| **38111.4**|
| Berzerk | 2339.0|**138167.9** |
| Bowling | 80.33|**84.3**|
| Breakout |**666.28**| 658.6|
| Enduro | 2359.25| **2363.3**|
| Freeway | **34.0**| **34.0** |
| Hero | **28564.58**| 27639.9|
| MontezumaRevenge| **5.55**| 0.6|
| Pong | **21.0**| **21.0**|
| PrivateEye |  **295.65**|289.9|
| Qbert | **29531.1**| 24950.3|
|Riverraid| **22519.31** | 20716.1 |
| Seaquest | **30870.0**| 27045.5|
| SpaceInvaders | **55127.5**|27810.9 |
| Venture | **719.51**|289.0 |
| VideoPinball | **731942.25**|664013.5 |

Double IQN wins on 12 domains!
There are 2 ties.
IQN wins on 4 domains.

File: chainerrl/agents/double_iqn.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+standard_library.install_aliases()  # NOQA
+
+import chainer.functions as F
+
+from chainerrl.agents import iqn
+
+
+class DoubleIQN(iqn.IQN):
+    """Double IQN.
+
```
Comment (muupan) [inline chainerrl/agents/double_iqn.py]: Can you add more explanations?

**Comment by muupan** [comment]
I think it is better to explicitly add `train_double_iqn.py` rather than `train_iqn_ale.py` because it will be clear from the filename what the file is for. IQN already has `examples/atari/reproduction/iqn`.

File: chainerrl/agents/double_iqn.py (hunk 1 above)
Comment (prabhatnagarajan) [reply chainerrl/agents/double_iqn.py]: Done.

File: chainerrl/agents/double_iqn.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import chainer.functions as F
+
+from chainerrl.agents import iqn
+
+
+class DoubleIQN(iqn.IQN):
+    """Double IQN. Using the primary Q-network's greedy/max action
+
+    to compute the target value rather than use the target network's
+    max action.
```
Comment (muupan) [inline chainerrl/agents/double_iqn.py]: ChainerRL generally uses Google style docstrings https://github.com/google/styleguide/blob/gh-pages/pyguide.md#381-docstrings

> A docstring should be organized as a summary line (one physical line) terminated by a period, question mark, or exclamation point, followed by a blank line, followed by the rest of the docstring starting at the same cursor position as the first quote of the first line.

I think it is not good to continue lines from the first line.

Comment (prabhatnagarajan) [reply chainerrl/agents/double_iqn.py]: Ok. Will change.

File: chainerrl/agents/double_iqn.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+standard_library.install_aliases()  # NOQA
+
+import chainer
+import chainer.functions as F
+
+from chainerrl.agents import iqn
+
+
+class DoubleIQN(iqn.IQN):
+    """Double IQN - Use primary network for target computation."""
```
Comment (muupan) [inline chainerrl/agents/double_iqn.py]: I don't think `Use primary network for target computation` is a good description of the algorithm. Can you just say something like `IQN with DoubleDQN-like target computation` and add a detailed explanation after a blank line (as suggested by the google style https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings)?

**Pull request merged**